src/communication/index.ts

```diff
@@ -1,108 +1,75 @@
-import {
-  InterfaceMessageIds,
-  InterfaceMessageEvent,
-  SentSDKMessage,
-  SDKMessageIds,
-  SDKMessageToPayload,
-  RequestId,
-  InterfaceMessageToPayload,
-  Communicator,
-} from '../types';
-import { INTERFACE_MESSAGES } from './messageIds';
-
-class InterfaceCommunicator implements Communicator {
+import semver from 'semver';
+import { InterfaceMessageEvent, Communicator, Methods, SDKRequestData } from '../types';
+import { generateRequestId, DEFAULT_ALLOWED_ORIGINS } from './utils';
+
+// eslint-disable-next-line
+type Callback = (response: any) => void;
+
+class PostMessageCommunicator implements Communicator {
   private allowedOrigins: RegExp[] = [];
+  private callbacks = new Map<string, Callback>();
+  sdkVersion: string;
 
-  constructor(allowedOrigins: RegExp[]) {
-    this.allowedOrigins = allowedOrigins;
+  constructor(allowedOrigins: RegExp[], sdkVersion: string) {
+    this.allowedOrigins = [...DEFAULT_ALLOWED_ORIGINS, ...allowedOrigins];
+    this.sdkVersion = sdkVersion;
 
     window.addEventListener('message', this.onParentMessage);
   }
 
-  private isValidMessage({ origin, data }: InterfaceMessageEvent): boolean {
-    const emptyOrMalformed = !data || !data.messageId;
+  private isValidMessage = ({ origin, data }: InterfaceMessageEvent): boolean => {
+    const emptyOrMalformed = !data;
     const unknownOrigin = this.allowedOrigins?.find((regExp) => regExp.test(origin)) === undefined;
     const sameOrigin = origin === window.origin;
+    const allowedSDKVersion = typeof data.version !== 'undefined' ? semver.gte(data.version, '1.0.0') : false;
 
-    return !emptyOrMalformed && !unknownOrigin && !sameOrigin;
-  }
+    return !emptyOrMalformed && !unknownOrigin && !sameOrigin && allowedSDKVersion;
+  };
 
-  private logIncomingMessage(origin: string, payload: InterfaceMessageToPayload[InterfaceMessageIds]): void {
-    console.info(`SafeConnector: A message was received from origin ${origin}. `, payload);
-  }
+  private logIncomingMessage = (msg: InterfaceMessageEvent): void => {
+    console.info(`Safe Apps SDK v2: A message was received from origin ${msg.origin}. `, msg.data);
```

Comment:
It is not really `v2`, but the first stable version :P

Reply:
how would you call it then? :P

Reply:
I would just call it `v1` ... before we did not have any version in the logs, so I think that would be fine ... theoretically we could even use the package version if this is easy, this way it is easy to debug what sdk is used.

Reply:
the exact version is included in messages